test/unittests/testutil.h

```diff
@@ -523,14 +549,57 @@ class MockCatalogManager : public AbstractCatalogManager<MockCatalog> {
                                  const shash::Any  &catalog_hash,
                                  MockCatalog *parent_catalog);
   MockCatalog* RetrieveRootCatalog() { return GetRootCatalog(); }
+  void AddFile(const DirectoryEntryBase &entry,
+               const XattrList &xattrs,
+               const std::string &parent_directory)
+  {
+    ++num_added_files_;
+  }
+
   void RegisterNewCatalog(MockCatalog *new_catalog) {
     string mountpoint_str = new_catalog->root_path();
     PathString mountpoint(mountpoint_str.c_str(), mountpoint_str.length());
     catalog_map_[mountpoint] = new_catalog;
   }
 
- protected:
+  unsigned GetNumAutogeneratedCatalogs() { return autogenerated_catalogs_; }
+  unsigned GetNumAddedFiles() { return num_added_files_; }
+
+  MockCatalog *FindCatalog(const PathString &path) {
+    map<PathString, MockCatalog*>::iterator it;
+    string best_match = "";
+    unsigned best_match_length = 0;
+    for (it = catalog_map_.begin(); it != catalog_map_.end(); ++it) {
+      string key = it->first.ToString();
+      if (path.ToString().find(key) != string::npos) {
+        unsigned length = key.length();
+        if (length > best_match_length) {
+          best_match_length = length;
+          best_match = key;
+        }
+      }
+    }
+    return catalog_map_[PathString(best_match)];
+  }
+
+  void CreateNestedCatalog(const string &mountpoint) {
```

Comment:
The think is that for testing purposes the CreateNestedCatalog method _should_ never be called, because it involves operations that cannot be unit tested. But yes, I agree: a comment would be good